synapse/models/infotech.py

```diff
@@ -25,559 +24,770 @@ class SemverType(DataType):
-
-def bruteStr(valu):
-    '''
-    Brute force the version out of a string.
-
-    Args:
-        valu (str): String to attempt to get version information for.
-
-    Notes:
-        This first attempts to parse strings using the it:semver normalization
-        before attempting to extract version parts out of the string.
-
-    Returns:
-        int, dict: The system normalized version integer and a subs dictionary.
-    '''
-    try:
-        valu, subs = s_datamodel.tlib.getTypeNorm('it:semver', valu)
-        return valu, subs
-    except s_common.BadTypeValu:
-        # Try doing version part extraction by noming through the string
-        subs = s_version.parseVersionParts(valu)
-        if subs is None:
-            raise s_common.BadTypeValu(valu=valu,
-                                       mesg='Unable to brute force version parts out of the string')
-        if subs:
-            valu = s_version.packVersion(subs.get('major'),
-                                         subs.get('minor', 0),
-                                         subs.get('patch', 0))
+class ItModule(s_module.CoreModule):
+    def initCoreModule(self):
+        self.model.form('it:dev:str').onAdd(self._onFormItDevStr)
+        self.model.form('it:dev:pipe').onAdd(self._onFormMakeDevStr)
+        self.model.form('it:dev:mutex').onAdd(self._onFormMakeDevStr)
+        self.model.form('it:dev:regkey').onAdd(self._onFormMakeDevStr)
+        self.model.prop('it:prod:softver:arch').onSet(self._onPropSoftverArch)
+        self.model.prop('it:prod:softver:vers').onSet(self._onPropSoftverVers)
+        self.model.prop('it:prod:softver:software').onSet(self._onPropSoftverSoft)
+
+    def bruteVersionStr(self, valu):
+        '''
+        Brute force the version out of a string.
+
+        Args:
+            valu (str): String to attempt to get version information for.
+
+        Notes:
+            This first attempts to parse strings using the it:semver normalization
+            before attempting to extract version parts out of the string.
+
+        Returns:
+            int, dict: The system normalized version integer and a subs dictionary.
+        '''
+        try:
+            valu, info = self.core.model.type('it:semver').norm(valu)
+            subs = info.get('subs')
             return valu, subs
+        except s_common.BadTypeValu:
+            # Try doing version part extraction by noming through the string
+            subs = s_version.parseVersionParts(valu)
+            if subs is None:
+                raise s_common.BadTypeValu(valu=valu,
+                                           mesg='Unable to brute force version parts out of the string')
+            if subs:
+                valu = s_version.packVersion(subs.get('major'),
+                                             subs.get('minor', 0),
+                                             subs.get('patch', 0))
+                return valu, subs
+
+    def _onFormItDevStr(self, node):
```

Comment:
Do these two functions also need to have oldv in their args?

Reply:
onForm callbacks online get the node